src/osd/OSD.h

```diff
@@ -1669,13 +1669,24 @@ class OSD : public Dispatcher,
  protected:
 
   // -- osd map --
-  // TODO: switch to std::atomic<OSDMapRef> when C++20 will be available.
-  OSDMapRef       _osdmap;
+#ifdef __cpp_lib_atomic_shared_ptr
+  std::atomic<OSDMapRef> _osdmap;
+#else
+  OSDMapRef _osdmap;
+#endif
   void set_osdmap(OSDMapRef osdmap) {
+#ifdef __cpp_lib_atomic_shared_ptr
+    _osdmap.store(osdmap);
+#else
```

Comment:
Do we actually need this? I thought `std::atomic_store` could work just as well on `std::atomic` as the `store`/`load` class methods.

Reply:
@adamemerson hi Adam, no, we don't. i agree with you that both of them compile and work. but in the light of #58881, i think we should be able to use a standard library with the proper support of `atomic<shared_ptr<...>>` very soon, by then, we will be able to use `_osdmap.store(osdmap)` without checking everywhere. IMHO, `osdmap.store(osdmap)` has better readability of `std::atomic_store(&_osdmap, osdmap)`. so, i'd prefer keeping this `#ifdef` pattern before #58881 lands. once we have the luxury of using a better standard library, we will be able to drop all these `#ifdef` checks and remove `std::atomic_store()` and friends in the `#else` branch in a cleanup pull request. what do you think?